Sources/Controllers/2.0/OACustomPOIViewController.mm

```diff
@@ -68,39 +99,77 @@ - (void)viewDidLoad
+}
 
-    self.saveButton.titleLabel.text = OALocalizedString(@"shared_string_save");
-    [self updateTextShowButton];
+- (void)updateShowButton:(BOOL)hasSelection
+{
+    self.showButton.backgroundColor = hasSelection ? UIColorFromRGB(color_primary_purple) : UIColorFromRGB(color_route_button_inactive);
+    [self.showButton setTintColor:hasSelection ? UIColor.whiteColor : UIColorFromRGB(color_text_footer)];
+    [self.showButton setTitleColor:hasSelection ? UIColor.whiteColor : UIColorFromRGB(color_text_footer) forState:UIControlStateNormal];
+    [self.showButton setUserInteractionEnabled:hasSelection];
+
+    if (_searchMode)
+    {
+        self.bottomView.hidden = YES;
+        self.showButton.hidden = YES;
+        self.tableBottomConstraint.constant = 0;
+    }
+    else
+    {
+        self.bottomView.hidden = NO;
+        self.showButton.hidden = NO;
+        self.tableBottomConstraint.constant = 76 + OAUtilities.getBottomMargin;
+    }
 }
 
 - (void)updateTextShowButton
 {
     _countShowCategories = 0;
-    for (OAPOICategory *category in _categories)
-        if ([_filter isTypeAccepted:category]) {
-            NSSet<NSString *> *acceptedSubtypes = [_filter getAcceptedSubtypes:category];
+    for (OAPOICategory *category in _acceptedTypes.keyEnumerator)
+    {
+        if ([_filter isTypeAccepted:category])
+        {
+            NSSet<NSString *> *acceptedSubtypes = [_acceptedTypes objectForKey:category];
             NSInteger count = acceptedSubtypes != [OAPOIBaseType nullSet] ? acceptedSubtypes.count : category.poiTypes.count;
             _countShowCategories += count;
         }
+    }
 
     NSString *textShow = OALocalizedString(@"sett_show");
     UIFont *fontShow = [UIFont systemFontOfSize:15 weight:UIFontWeightSemibold];
-    UIColor *colorShow = [[UIColor alloc] initWithWhite:1 alpha:1];
-    NSMutableAttributedString *attrShow = [[NSMutableAttributedString alloc] initWithString:textShow attributes:@{NSFontAttributeName:fontShow, NSForegroundColorAttributeName:colorShow}];
+    UIColor *colorShow = _countShowCategories != 0 ? UIColor.whiteColor : UIColorFromRGB(color_text_footer);
+    NSMutableAttributedString *attrShow = [[NSMutableAttributedString alloc] initWithString:textShow attributes:@{NSFontAttributeName: fontShow, NSForegroundColorAttributeName: colorShow}];
 
     NSString *textCategories = [NSString stringWithFormat:@"\n%@: %li", OALocalizedString(@"categories"), _countShowCategories];
     UIFont *fontCategories = [UIFont systemFontOfSize:13 weight:UIFontWeightRegular];
-    UIColor *colorCategories = [[UIColor alloc] initWithWhite:1 alpha:0.5];
-    NSMutableAttributedString *attrCategories = [[NSMutableAttributedString alloc] initWithString:textCategories attributes:@{NSFontAttributeName:fontCategories, NSForegroundColorAttributeName:colorCategories}];
+    UIColor *colorCategories = _countShowCategories != 0 ? [[UIColor alloc] initWithWhite:1 alpha:0.5] : UIColorFromRGB(color_text_footer);
```

Comment:
Is [UIColor alloc] initWithWhite:1 alpha:0.5] from Figma?

Reply:
Yes